lib/block_view.go

```diff
@@ -4568,17 +4568,6 @@ func (bav *UtxoView) ConnectBlock(
 	// after connecting all transactions in the block. These operations
 	// are always the last utxo operation in a given block.
 	var blockLevelUtxoOps []*UtxoOperation
-	if blockHeight >= uint64(bav.Params.ForkHeights.BalanceModelBlockHeight) &&
```

Comment:
Deleting expired nonces significantly slows down the syncing process. Eventually we can do it on some different cadence or look at performance enhancements for this iteration.

Reply:
Yep sounds fine to me. Maybe put a TODO here to delete nonces and explain the whole situation for posterity. It's always good to throw in gratuitous comments to capture the tribal knowledge. Something like the following:

```
TODO: To prevent the state from bloating, we should delete nonces periodically. We used to do that here but it was causing badger seeks to be slow due to a bug in badger whereby deleting keys slows down seeks. Eventually, we should go back to deleting txn nonces if we fix that badger bug or find a workaround.
```